source/AI.cpp

```diff
@@ -2307,7 +2308,9 @@ bool AI::DoCloak(Ship &ship, Command &command)
 		// or 40% farther away before it begins decloaking again.
 		double hysteresis = ship.Commands().Has(Command::CLOAK) ? .4 : 0.;
 		// If cloaking costs nothing, and no one has asked you for help, cloak at will.
-		bool cloakFreely = !attributes.Get("cloaking fuel") && !ship.GetShipToAssist();
+		bool cloakFreely = 0;
```

Comment:
Why did you change the structure of the original check? (And why initialize a `bool` with an `int` when the literals `true` and `false` exist and their use ensures your intent is explicitly clear?)

`bool cloakFreely = fuelCost <= 0. && !ship.GetShipToAssist()`